modules/monitoring-resolving-the-kubepersistentvolumefillingup-alert-firing-for-prometheus.adoc

```diff
@@ -0,0 +1,79 @@
+// Module included in the following assemblies:
+//
+// * monitoring/troubleshooting-monitoring-issues.adoc
+// * support/troubleshooting/investigating-monitoring-issues.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="resolving-the-kubepersistentvolumefillingup-alert-firing-for-prometheus_{context}"]
+= Resolving the KubePersistentVolumeFillingUp alert firing for Prometheus
+
+As a cluster administrator, you can resolve the `KubePersistentVolumeFillingUp` alert being triggered for Prometheus. 
+
+The critical alert fires when a persistent volume (PV) claimed by a `prometheus-k8s-*` pod in the `openshift-monitoring` project has less than 3% of its total space remaining. This can cause Prometheus to not function normally.
+
+[NOTE]
+====
+There are two `KubePersistentVolumeFillingUp` alerts:
+
+* *Critical alert*:  The alert with the `severity="critical"` label is triggered when the mounted PV has less than 3% of its total space remaining.
+* *Warning alert*: The alert with the `severity="warning"` label is triggered when the mounted PV has less than 15% of its total space remaining and is expected to fill up within four days.
+====
+
+To address this issue, you can remove Prometheus time series database (TSDB) blocks to create more space for the PV.
+
+.Prerequisites
+
+ifndef::openshift-dedicated,openshift-rosa[]
+* You have access to the cluster as a user with the `cluster-admin` cluster role.
+endif::openshift-dedicated,openshift-rosa[]
+ifdef::openshift-dedicated,openshift-rosa[]
+* You have access to the cluster as a user with the `dedicated-admin` role.
+endif::openshift-dedicated,openshift-rosa[]
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. List the size of all TSDB blocks, sorted from oldest to newest:
++
+[source,terminal]
+----
+$ oc debug <prometheus_k8s_pod_name> -n openshift-monitoring \// <1>
+-c prometheus --image=$(oc get po -n openshift-monitoring <prometheus_k8s_pod_name> \// <1>
+-o jsonpath='{.spec.containers[?(@.name=="prometheus")].image}') \
+-- sh -c 'cd /prometheus/;du -hs $(ls -dt */ | grep -Eo "[0-9|A-Z]{26}")'
+----
+<1> Replace `<prometheus_k8s_pod_name>` with the pod mentioned in the `KubePersistentVolumeFillingUp` alert description.
```

Comment:
Hello @juzhao could you please help me with the output for this command?
With my simple testing I only got very simple output that looked like this:
```
Starting pod/prometheus-k8s-0-debug-7hjzp ...
72M	.
```
Would you help me provide more complex output that would be close to real example? I think that would be very beneficial to include.
Thank you very much in advance!

Reply:
```
308K    01HVKMPKQWZYWS8WVDAYQHNMW6
5.2M    01HVK64DTDA81799TBR9QDECEZ
1.2M    01HVK64DS7TRZRWF2756KHST5X
140K    01HVJS59K11FBVAPVY57K88Z11
9.0M    01HVH2A5Z58SKT810EM6B9AT50
1.2M    01HV8ZDVQMX41MKCN84S32RRZ1
3.4M    01HV6Q2N26BK63G4RYTST71FBF
156K    01HV664H9J9Z1FTZD73RD1563E
16K     01HTHXB60A7F239HN7S2TENPNS
104K    01HTHMGRXGS0WXA3WATRXHR36B
````

Reply:
Simon, thank you very much for the output! :heart: